corehq/motech/repeaters/models.py

```diff
@@ -221,6 +221,8 @@ def __new__(cls, *args, **kwargs):
                 cls._meta.get_field(cls.PROXY_FIELD_NAME))
             try:
                 proxy_class_name = args[proxy_name_field_index]
+                if isinstance(proxy_class_name, Deferred):
+                    return super().__new__(repeater_class)
```

Comment:
This makes me a little nervous. It means there are times when this will return a `SQLRepeater` when it should be returning something else. In the case of `TestSQLRepeaterConnectionSettings.test_used_connection_setting_cannot_be_deleted` it hits this line because it is loading related repeaters of a `ConnectionSettings` instance, and it only queries ids from the database and sets all other fields to `DEFERRED`. That works fine for that test, but what other situations could that happen?

Reply:
We have fair cover for all CRUD operations that we perform on repeaters. They all seem to work fine with it. So this gives us a bit of confidence. In normal operations that we use things should run fine.

Another reason why I feel this should be okay is whenever we are getting deferred fields then we are defaulting to Django's default behavior by returning `super().__new__(repeater_class)` where `repeater_class` equals `cls` passed into `__new__`.